### PR TITLE
Where #2634

### PR DESCRIPTION
**Pull request by Amri-Tarik**
issue : #2634
Implementing numpy's "where" function (indexing_routines) in ivy's frontend.
any criticisms or tips are welcome.

### CODE REVIEW

**Comment by Amri-Tarik** [comment]
@juliagsy can you please check what's wrong with the GitHub actions tests ? they seem to be failing in functions unrelated to what i have edited (like add or substract).
`test_numpy_where()` seems to be working fine locally for me when using Pycharm remote interpreter with docker on latest image (downloaded today).

**Comment by juliagsy** [comment]
Hello! No worries on that! Some functions are failing on master so it might be unrelated to your changes. Also, I'll review as soon as possible! Thanks for contributing!

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hello there! Great work so far! Please rename the files/imports so that they match conventions from NumPy - https://numpy.org/doc/stable/reference/routines.sort.html

Main files: indexing_routines >> sorting_searching_counting
subfiles: indexing_routines.py >> searching.py
(for tests, please update to the same names but with `test_` as prefix of every files)

Once you're done, please request my review again, thank you!

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py [hunk 1]
```diff
@@ -0,0 +1,60 @@
+from datetime import timedelta
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py:1]: unused import

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+
+import hypothesis.extra.numpy as hnp
+import numpy as np
+from hypothesis import given, strategies as st, settings
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py:5]: `settings` is unused import

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+from datetime import timedelta
+
+import hypothesis.extra.numpy as hnp
+import numpy as np
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py:4]: unused import

File: ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+
+    shapes_st = hnp.mutually_broadcastable_shapes(num_shapes=3, min_dims=1, min_side=1)
+    cond_shape, x1_shape, x2_shape = draw(shapes_st).input_shapes
+    cond = draw(hnp.arrays(hnp.boolean_dtypes(), cond_shape))
+    x1 = draw(hnp.arrays(dtype, x1_shape))
+    x2 = draw(hnp.arrays(dtype, x2_shape))
+    return cond, x1, x2, dtype
+
+
+
```
Comment (juliagsy) [inline ivy_tests/test_ivy/test_frontends/test_numpy/test_sorting_searching_counting/test_searching.py:25]: for lint fix, remove a blank line (keep 2 blank lines only)

File: ivy/functional/frontends/numpy/sorting_searching_counting/searching.py [hunk 5]
```diff
@@ -0,0 +1,12 @@
...
+
+
+def where(cond, x1=None, x2=None, /):
+    if x1 is None and x2 is None :
+        # numpy where behaves as np.asarray(condition).nonzero() when x and y not included
+        return ivy.asarray(cond).nonzero()
+    elif x1 is not None and x2 is not None :
+        return ivy.where(cond, x1, x2)
+    else :
+        raise TypeError(f'where takes either 1 or 3 arguments')
```
Comment (juliagsy) [inline ivy/functional/frontends/numpy/sorting_searching_counting/searching.py:12]: lint fix - I think you can directly use double quoting as string placeholder or formatting isn't required

**Review by juliagsy — CHANGES_REQUESTED** [review]
Hey! Looks good so far! Only some issues related to lint fixes and unused imports, please request my review when you're done and thanks for contributing!

**Comment by Amri-Tarik** [comment]
sorry for giving you some extra work, I was too focused trying to pass the tests I forgot to clean the code.
this should hopefully be the last one, thanks for the review and the tips.

**Comment by juliagsy** [comment]
No worries! Thanks for all the changes, will merge in soon!

**Review by juliagsy — APPROVED** [review]
Thanks for contributing! I'll merge the PR soon!

**Pull request merged**